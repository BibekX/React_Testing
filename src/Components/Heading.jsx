import React from "react";

export default function Heading() {
  return (
    <div>
      <h1>Heading</h1>
      <p>Learn Angular and React</p>
      <button>Click Me</button>
      <input type="text" placeholder="username" />
      <p>Learn Angular and React</p>
    </div>
  );
}
