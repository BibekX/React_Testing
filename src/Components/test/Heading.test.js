import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Heading from "../Heading";

test("check the snapshot", () => {
  let tree = renderer.create(<Heading />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("check if Learn Angular exists in Heading", async () => {
  //Arrange
  render(<Heading />);

  //Act
  const headingElement = screen.queryAllByText(/^learn angular/i);
  const headingElement2 = screen.getByRole("heading");
  const headingElement3 = screen.getByPlaceholderText("username");

  //Assert
  expect(headingElement).toBeInTheDocument();
  expect(headingElement2).toBeInTheDocument();
  expect(headingElement3).toBeInTheDocument();

  // FIRE EVENT
  const inputElement = screen.getByPlaceholderText("username");
  fireEvent.change(inputElement, { target: { value: "Brock" } });
  expect(inputElement.value).toBe("Brock");
});
