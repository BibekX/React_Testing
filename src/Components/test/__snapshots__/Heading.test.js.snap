// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the snapshot 1`] = `
<div>
  <h1>
    Heading
  </h1>
  <p>
    Learn Angular
  </p>
</div>
`;
